lightkurve/correctors/designmatrix.py

```diff
@@ -379,44 +429,396 @@ def __getitem__(self, key):
+
+    def to_dense(self):
+        return DesignMatrix(self.values, name=self.name,
+                            columns=self.columns, prior_mu=self.prior_mu,
+                             prior_sigma=self.prior_sigma)
+
+
+class SparseDesignMatrixCollection(DesignMatrixCollection):
+    """A set of design matrices."""
+    def __init__(self, matrices, warn=True):
+        if not np.all([issparse(m.X) for m in matrices]):
+            if warn:
+                warnings.warn(('Not all matrices are `SparseDesignMatrix` objects. '
+                                'Dense matrices will be converted to sparse matrices.'), LightkurveWarning)
+            sparse_matrices = []
+            for m in matrices:
+                if isinstance(m, DesignMatrix):
+                    sparse_matrices.append(m.copy().to_sparse())
+                else:
+                    sparse_matrices.append(m)
+            self.matrices = sparse_matrices
+        else:
+            self.matrices = matrices
+        self.X = hstack([m.X for m in self.matrices], format='csr')
+        self._child_class = SparseDesignMatrix
+        self.validate()
+
+    def plot(self, ax=None, **kwargs):
+        """Visualize the design matrix values as an image.
+
+        Uses Matplotlib's `~lightkurve.utils.plot_image` to visualize the
+        matrix values.
+
+        Parameters
+        ----------
+        ax : `~matplotlib.axes.Axes`
+            A matplotlib axes object to plot into. If no axes is provided,
+            a new one will be created.
+        **kwargs : dict
+            Extra parameters to be passed to `.plot_image`.
+
+        Returns
+        -------
+        `~matplotlib.axes.Axes`
+            The matplotlib axes object.
+        """
+        temp_dm = SparseDesignMatrix(hstack([d.X for d in self]))
+        ax = temp_dm.plot(**kwargs)
+        ax.set_title("Design Matrix Collection")
+        return ax
+
+    def __repr__(self):
+        return 'SparseDesignMatrixCollection:\n' + \
+                    ''.join(['\t{}\n'.format(i.__repr__()) for i in self])
+
+
+
 ####################################################
 # Functions to create commonly-used design matrices.
 ####################################################
 
-def create_spline_matrix(x, n_knots=20, degree=3, name='spline',
-                         include_intercept=False):
-    """Returns a `.DesignMatrix` which models splines using `patsy.dmatrix`.
 
+@jit(nopython=True)
+def basis(x, degree, i, knots):
+    """Create a spline basis vector for an input x, for the ith knot.
+    Parameters
+    ----------
+    x : np.ndarray
+        Input x
+    degree : int
+        Degree of spline to calculate basis for
+    i : int
+        The index of the knot to calculate the basis for
+    knots : np.ndarray
+        Array of all knots
+    Returns
+    -------
+    B : np.ndarray
+        A vector of same length as x containing the spline basis for the ith knot
```

Comment:
Adding a short `Examples` section to the docstring could be really great for these spline functions.

Reply:
It's hard to add an example here because this is a recursive function, what sort of example were you thinking?

Reply:
Even though it's not a public function, I think it's important to add a few more words to the docstring, e.g. a link to the equation (e.g. Wikipedia) that's being implemented, or a short example showing what it returns and why.